src/vehicle.cpp

```diff
@@ -2437,10 +2442,14 @@ item_location vehicle::part_base( int p )
 
 int vehicle::find_part( const item &it ) const
 {
-    auto idx = std::find_if( parts.begin(), parts.end(), [&it]( const vehicle_part & e ) {
-        return &e.base == &it;
-    } );
-    return idx != parts.end() ? std::distance( parts.begin(), idx ) : INT_MIN;
+    int index = INT_MIN;
```

Comment:
> 
> 
> Would you please resubmit the vehicle::parts privatization as a separate PR? It's a pretty straightforward set of commits, but it has a lot of impact on future development and is going to be the source of a lot of merge conflicts. Get it out of the way as a separate PR, and then evaluating this PR is going to be easier.

Yes. I think that would make a lot more sense. The code is fairly robust since I've tested it extensively while implementing fake parts. And as you say, it's the code that is causing the merge conflicts. I will look at doing it later today - there is a rebase between the encapsulation parts and this PR in my branch, so will have to have a new look at veh_interact.cpp.